homeassistant/components/cookidoo/config_flow.py

```diff
@@ -0,0 +1,208 @@
+        """Allow reconfiguration of localization and ignore auth."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            if not (
+                errors := await self.validate_input(
+                    {**self.reconfigure_entry.data, **user_input}
+                )
+            ):
+                return self.async_update_reload_and_abort(
+                    self.reconfigure_entry, data_updates=user_input
+                )
+        return self.async_show_form(
+            step_id="reconfigure_confirm",
+            data_schema=self.add_suggested_values_to_schema(
+                data_schema=vol.Schema(self.LOCALIZATION_DATA_SCHEMA),
+                suggested_values=self.reconfigure_entry.data,
+            ),
+            description_placeholders={CONF_NAME: self.reconfigure_entry.title},
+            errors=errors,
+        )
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Perform reauth upon an API authentication error."""
+        await self.generate_schemata()
+        self.reauth_entry = self._get_reauth_entry()
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Dialog that informs the user that reauth is required."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            if not (
+                errors := await self.validate_input(
+                    {**self.reauth_entry.data, **user_input}
+                )
+            ):
+                return self.async_update_reload_and_abort(
+                    self.reauth_entry, data_updates=user_input
+                )
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=self.add_suggested_values_to_schema(
+                data_schema=vol.Schema(self.AUTH_DATA_SCHEMA),
+                suggested_values={CONF_EMAIL: self.reauth_entry.data[CONF_EMAIL]},
+            ),
+            description_placeholders={CONF_NAME: self.reauth_entry.title},
+            errors=errors,
+        )
+
+    async def generate_schemata(self) -> None:
+        """Generate schemata."""
+        self.localizations = await get_localization_options()
+
+        self.AUTH_DATA_SCHEMA = {
+            vol.Required(CONF_EMAIL): TextSelector(
+                TextSelectorConfig(
+                    type=TextSelectorType.EMAIL,
+                    autocomplete="email",
+                ),
+            ),
+            vol.Required(CONF_PASSWORD): TextSelector(
+                TextSelectorConfig(
+                    type=TextSelectorType.PASSWORD,
+                    autocomplete="current-password",
+                ),
+            ),
+        }
+
+        self.LOCALIZATION_DATA_SCHEMA = {
+            vol.Required(
+                CONF_LOCALIZATION, default=DEFAULT_LOCALIZATION
```

Comment:
I want to avoid a two step flow and not every country allows every language.

Reply:
I did have it in a single step before, but it gave a lot of invalid combination. I can convert it to a two step flow if needed, but it would complicate it for not much benefit?

Reply:
in a 2-step flow you could then offer only the valid options based on the selection from the previous step. But it would also be good for a later improvement.

Reply:
I'd prefer to keep it simple for a first PR and do such QoL changes later on